gnomad_qc/v3/create_release/prepare_vcf_data_release.py

```diff
@@ -0,0 +1,1019 @@
+
+    # Reformat names to remove "adj" pre-export
+    # e.g, renaming "AC-adj" to "AC"
+    # All unlabeled frequency information is assumed to be adj
+    logger.info("Dropping 'adj' from info annotations...")
+    row_annots = list(ht.info)
+    new_row_annots = []
+    for x in row_annots:
+        x = x.replace("-adj", "")
+        x = x.replace("-", "_")  # VCF 4.3 specs do not allow hyphens in info fields
+        new_row_annots.append(x)
+
+    info_annot_mapping = dict(
+        zip(new_row_annots, [ht.info[f"{x}"] for x in row_annots])
+    )
+    ht = ht.transmute(info=hl.struct(**info_annot_mapping))
+
+    logger.info("Adjusting VCF incompatiable types...")
+    # Reformat AS_SB_TABLE for use in ht_to_vcf_mt
+    ht = ht.annotate(
+        info=ht.info.annotate(
+            AS_SB_TABLE=hl.array([ht.info.AS_SB_TABLE[:2], ht.info.AS_SB_TABLE[2:]])
+        )
+    )
+    ht = adjust_vcf_incompatible_types(ht)
+
+    return ht, new_row_annots
+
+
+def build_parameter_dict(
+    hgdp_1kg_subset: bool = False, test: bool = False
+) -> Dict[str, Union[bool, str, List, Dict, Set, hl.Table, None]]:
+    """
+    Build a dictionary of the parameters needed in the script because they differ for the HGDP + TGP subset release
+    compared to the full release.
+
+    :param hgdp_1kg_subset: Build the parameter list specific to the HGDP + TGP subset release.
+    :param test: Uses a checkpoint path for the prepared VCF Table that is specific to testing.
+    :return: Dictionary containing parameters needed for making the release VCF.
+    """
+    if hgdp_1kg_subset:
+        parameter_dict = {
+            "pops": HGDP_KG_POPS,
+            "subsets": ["", "gnomad"],
+            "is_subset": True,
+            "temp_ht_path": get_checkpoint_path(
+                f"vcf_prep_{'test' if test else ''}_hgdp_1kg"
+            ),
+            "include_age_hists": False,
+            "subset_pops": {"gnomad": POPS, "": HGDP_KG_POPS},
+            "vcf_info_reorder": HGDP_KG_VCF_INFO_REORDER,
+            "ht": release_subset(subset="hgdp_1kg", dense=True).mt().rows(),
+            "freq_entries_to_remove": set(),
+            "age_hist_data": None,
+        }
+
+    else:
+        parameter_dict = {
+            "pops": POPS,
+            "subsets": SUBSET_LIST_FOR_VCF,
+            "is_subset": False,
+            "temp_ht_path": get_checkpoint_path(
+                f"vcf_prep_{'test' if args.test else ''}"
+            ),
+            "include_age_hists": True,
+            "subset_pops": {"hgdp": HGDP_POPS, "tgp": KG_POPS},
+            "vcf_info_reorder": VCF_INFO_REORDER,
+            "ht": hl.read_table(
+                release_ht_path()
+            ),  # TODO: Change to release_sites().ht(),
+        }
+        # Downsampling and subset entries to remove from VCF's freq export
+        # Note: Need to extract the non-standard downsamplings from the freq_meta struct to the FREQ_ENTRIES_TO_REMOVE
```

Comment:
If a population has less samples than one of the down sampling numbers it will get downsampled to it's max sample size instead of one of the standard downsamplings in the `DOWNSAMPLINGS` global